### DISCUSSION TITLE
MTEX 5.11.1  / NFFT 3.5.3 installation on macOS 14.4.1 (Sonoma)

### DISCUSSION THREAD

**Discussion by tchufnagel** [mtex-toolbox/mtex#2120] [Ask Anything]
I am trying to get MTEX 5.11.1 running on macOS 14.4.1 on Apple M1 machine. My original attempt was using the 'startup_mtex' script in Matlab, but it failed in trying to run NFFT. I downloaded NFFT 3.5.3 and compiled it, but trying to configure it throws an error: 

`configure: error: You don't seem to have a C compiler with OpenMP support installed which is required for threaded NFFT`

Online searches don't provide much help, but the gist of them seems to be to install a different C compiler and edit the configuration files to point at it rather than gcc.

Is that really necessary, or am I missing something? I'm thinking I may have wandered off the garden path...

**Comment by lfgmorales** [comment]
Hi

Do you have xcode installed in your mac?
Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 18:45, tchufnagel ***@***.***> wrote:

﻿

I am trying to get MTEX 5.11.1 running on macOS 14.4.1 on Apple M1 machine. My original attempt was using the 'startup_mtex' script in Matlab, but it failed in trying to run NFFT. I downloaded NFFT 3.5.3 and compiled it, but trying to configure it throws an error:

configure: error: You don't seem to have a C compiler with OpenMP support installed which is required for threaded NFFT

Online searches don't provide much help, but the gist of them seems to be to install a different C compiler and edit the configuration files to point at it rather than gcc.

Is that really necessary, or am I missing something? I'm thinking I may have wandered off the garden path...

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4I3LHOEV7W5QNUDY3Y572CNAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGUZDKMBXGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by tchufnagel** [comment]
Yes, I do.

—-
Todd Hufnagel, Professor
Materials Science and Engineering
Johns Hopkins University

On Apr 18, 2024, at 12:58, Luiz Morales ***@***.***> wrote:

﻿

Hi

Do you have xcode installed in your mac?
Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 18:45, tchufnagel ***@***.***> wrote:

﻿

I am trying to get MTEX 5.11.1 running on macOS 14.4.1 on Apple M1 machine. My original attempt was using the 'startup_mtex' script in Matlab, but it failed in trying to run NFFT. I downloaded NFFT 3.5.3 and compiled it, but trying to configure it throws an error:

configure: error: You don't seem to have a C compiler with OpenMP support installed which is required for threaded NFFT

Online searches don't provide much help, but the gist of them seems to be to install a different C compiler and edit the configuration files to point at it rather than gcc.

Is that really necessary, or am I missing something? I'm thinking I may have wandered off the garden path...

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4I3LHOEV7W5QNUDY3Y572CNAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGUZDKMBXGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by lfgmorales** [comment]
Try here

<https://mtex-toolbox.github.io/installation>
Installation Issues for Mac OSX | MTEX<https://mtex-toolbox.github.io/installation>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/installation>
        [favicon.png] <https://mtex-toolbox.github.io/installation>


Compiling the nfft

Maybe helps

Cheers

L

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 19:02, tchufnagel ***@***.***> wrote:

﻿

Yes, I do.

—-
Todd Hufnagel, Professor
Materials Science and Engineering
Johns Hopkins University

On Apr 18, 2024, at 12:58, Luiz Morales ***@***.***> wrote:

﻿

Hi

Do you have xcode installed in your mac?
Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 18:45, tchufnagel ***@***.***> wrote:

﻿

I am trying to get MTEX 5.11.1 running on macOS 14.4.1 on Apple M1 machine. My original attempt was using the 'startup_mtex' script in Matlab, but it failed in trying to run NFFT. I downloaded NFFT 3.5.3 and compiled it, but trying to configure it throws an error:

configure: error: You don't seem to have a C compiler with OpenMP support installed which is required for threaded NFFT

Online searches don't provide much help, but the gist of them seems to be to install a different C compiler and edit the configuration files to point at it rather than gcc.

Is that really necessary, or am I missing something? I'm thinking I may have wandered off the garden path...

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4I3LHOEV7W5QNUDY3Y572CNAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGUZDKMBXGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157562>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFBZBTW4BKUXNK5ALP7AZ6LY573RVAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNJXGU3DE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN6THB6MI6CJ5BD3RTDY574CFAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNJXGYYDQ>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by tchufnagel** [comment]
Thanks, yes, I am aware of that page and have been trying to work my way through it. But the instructions for compiling nifft provided there do not work on my machine (trying to run the bootstrap script throws errors) which is why I have been trying to follow the instructions provided at the nfft site (http://www.fftw.org/fftw3_doc/Installation-on-Unix.html).

Todd

------
Todd Hufnagel
Professor
Department of Materials Science and Engineering
Johns Hopkins University

On Apr 18, 2024, at 1:13 PM, Luiz Morales ***@***.***> wrote:


Try here

<https://mtex-toolbox.github.io/installation>
Installation Issues for Mac OSX | MTEX<https://mtex-toolbox.github.io/installation>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/installation>
[favicon.png] <https://mtex-toolbox.github.io/installation>


Compiling the nfft

Maybe helps

Cheers

L

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 19:02, tchufnagel ***@***.***> wrote:

﻿

Yes, I do.

—-
Todd Hufnagel, Professor
Materials Science and Engineering
Johns Hopkins University

On Apr 18, 2024, at 12:58, Luiz Morales ***@***.***> wrote:

﻿

Hi

Do you have xcode installed in your mac?
Cheers

Luiz
Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 18 Apr 2024, at 18:45, tchufnagel ***@***.***> wrote:

﻿

I am trying to get MTEX 5.11.1 running on macOS 14.4.1 on Apple M1 machine. My original attempt was using the 'startup_mtex' script in Matlab, but it failed in trying to run NFFT. I downloaded NFFT 3.5.3 and compiled it, but trying to configure it throws an error:

configure: error: You don't seem to have a C compiler with OpenMP support installed which is required for threaded NFFT

Online searches don't provide much help, but the gist of them seems to be to install a different C compiler and edit the configuration files to point at it rather than gcc.

Is that really necessary, or am I missing something? I'm thinking I may have wandered off the garden path...

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4I3LHOEV7W5QNUDY3Y572CNAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGUZDKMBXGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157562>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFBZBTW4BKUXNK5ALP7AZ6LY573RVAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNJXGU3DE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN6THB6MI6CJ5BD3RTDY574CFAVCNFSM6AAAAABGNUUOQ2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNJXGYYDQ>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608", "url": "https://github.com/mtex-toolbox/mtex/discussions/2120#discussioncomment-9157608", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by ralfHielscher** [comment]
MTEX 5.11.2 is out and ready for testing 

https://github.com/mtex-toolbox/mtex/releases/download/mtex-5.11.2/mtex-5.11.2.zip

It includes new binaries for intel and arm macs.
There is no automatic check at startup. You may want to type

check_mex
to get a detailed overview about which mex files are working and which not.
You can do

mex_install force
to recompile the mex-files on your own computer.

Any feedback is appreciated.
Ralf.
